fedimint-core/src/bitcoin_migration.rs

```diff
@@ -110,6 +110,15 @@ pub fn bitcoin30_to_bitcoin32_payment_preimage(
     lightning_types::payment::PaymentPreimage(preimage.0)
 }
 
+pub fn bitcoin30_to_bitcoin32_sha256_hash(
+    hash: &bitcoin30::hashes::sha256::Hash,
+) -> bitcoin::hashes::sha256::Hash {
+    bincode::deserialize(
+        &bincode::serialize(&hash).expect("Failed to serialize bitcoin30 sha256 hash"),
+    )
+    .expect("Failed to convert bitcoin30 sha256 hash to bitcoin32 sha256 hash")
```

Comment:
That's the third creative way you have found to transform hashes in rust-bitcoin :laughing: 

When you look at the [`sha256::Hash` impl](https://docs.rs/bitcoin_hashes/latest/bitcoin_hashes/sha256/struct.Hash.html) there are methods to converting from/to byte array references:

![Screenshot 2024-10-24 at 10-22-57 Hash in bitcoin_hashes sha256 - Rust](https://github.com/user-attachments/assets/049f2fb5-fd38-46c2-8f9e-c8fba88461d4)
![Screenshot 2024-10-24 at 10-22-49 Hash in bitcoin_hashes sha256 - Rust](https://github.com/user-attachments/assets/8a4a1c89-8ccc-405c-b320-149edf3506e1)

Reply:
Haha achievement unlocked...?

Fixed. Could probably optimize some of the existing migration functions, but will do that in a follow-up.

Reply:
Idk if necssary to touch other fns just yet, ideally we will get rid of them soonish anyway. Just introducing new ones seems very unnecessary. If we get stuck for some reason I'd do the refactor.

Reply:
> Fixed

Did you push?

Reply:
> Did you push?

Whoops! Pushed.